lib/client.js

```diff
@@ -91,43 +92,52 @@ function _connect (client) {
             err = new Error('aborted')
           }
 
-          callback()
-
           cb(err, null)
         }
       })
-      body.push = request.wrapSimple(body, body.push)
+      _body.push = request.wrapSimple(_body, _body.push)
     }
 
     request.callback(null, {
       statusCode,
       headers: parseHeaders(headers),
-      body
+      body: _body
     })
 
     if (skipBody) {
       callback()
+    } else {
+      _callback = callback
     }
 
     return skipBody
   }
 
   parser[HTTPParser.kOnBody] = (chunk, offset, length) => {
-    read += length
+    _read += length
 
-    if (body.destroyed) {
-      if (read > client.maxAbortedPayload) {
+    if (_body.destroyed) {
+      if (_read > client.maxAbortedPayload) {
         socket.destroy()
       }
-    } else if (!body.push(chunk.slice(offset, offset + length))) {
+    } else if (!_body.push(chunk.slice(offset, offset + length))) {
       socket.pause()
     }
   }
 
   parser[HTTPParser.kOnMessageComplete] = () => {
-    if (body) {
-      body.push(null)
-      body = null
+    if (_body) {
+      // Stop Readable from emitting 'end' when destroyed.
+      if (!_body.destroyed) {
+        _body.push(null)
+      }
```

Comment:
Without this we can get an `'end'` after having aborted the response which is surprising